src/protocol/dp/dp_budgets.rs

```diff
@@ -0,0 +1,48 @@
+// DP budgets
+
+pub struct Epoch {
+    epoch_id: u32,
+    epoch_budget: u32,
+}
+
+pub struct SiteBudget {
+    pub domain: String,
+    //pub public_key: ??,
+
+    // suppose we store budget for 5 epochs
+    // TODO: probably better to store as an array
```

Comment:
Yes, though you might maintain a list of (week, budget) instead.  It might be better that way to avoid relying on any conception of what "today" or "this week" is.  Timeless code is always better.